vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -252,25 +252,51 @@ def _get_unfiltered_data(
 def _get_modified_page_figures(
     ctds_filter: list[CallbackTriggerDict],
     ctds_filter_interaction: list[dict[str, CallbackTriggerDict]],
-    ctds_parameters: list[CallbackTriggerDict],
+    ctds_parameter: list[CallbackTriggerDict],
     targets: list[ModelID],
 ) -> dict[ModelID, Any]:
+    from vizro.models import Filter
+
     outputs: dict[ModelID, Any] = {}
 
+    control_targets = []
+    control_targets_targets = []
+    figure_targets = []
+    for target in targets:
+        target_obj = model_manager[target]
+        if isinstance(target_obj, Filter):
+            control_targets.append(target)
+            control_targets_targets.extend(target_obj.targets)
+        else:
+            figure_targets.append(target)
+
+    # Get data_frames for all figure_targets and components that are targets of controls
+    _get_unfiltered_data_targets = list(set(figure_targets + control_targets_targets))
+
+    figure_targets_unfiltered_data: dict[ModelID, pd.DataFrame] = _get_unfiltered_data(
+        ctds_parameter=ctds_parameter, targets=_get_unfiltered_data_targets
+    )
+
```

Comment:
Most of this code can be removed. It's here only to support future cases like when the dynamic Filter is the direct target of a Parameter.

Retrieving only action's "targets" data_frames from `_multi_load` is not the best solution. In that way, we assume that `Filter.targets` are the part of the `action's targets` which may not need to be true. It would work for the `on_page_load` as all figures on the right side of the screen are indeed action's targets, but it will not work if targets are explicitly set for some other action. For example, in future, if Parameter is targeting only a single Filter.

Reply:
Done in -> https://github.com/mckinsey/vizro/pull/879/commits/161a1e9e34c6c6032f9ff3c3081ec35ffdb262a1

Reply:
Thank you, that's much clearer now!